lib/modules/platform/bitbucket-server/utils.ts

```diff
@@ -155,38 +157,80 @@ export function getInvalidReviewers(err: BitbucketError): string[] {
   return invalidReviewers;
 }
 
+function generateUrlFromEndpoint(
+  defaultEndpoint: string,
+  opts: HostRule,
+  repository: string
+): string {
+  const { host, pathname } = url.parse(defaultEndpoint);
+  const generatedUrl = git.getUrl({
+    protocol: defaultEndpoint.split(':')[0] as GitProtocol,
```

Comment:
why not using from parsed url?

Reply:
Good point, I also changed the deprecated `url.parse(defaultEndpoint);` for `new URL(defaultEndpoint);`

Reply:
use `parseUrl` from `util/url`

Reply:
and please do the refactoring / preparation in a separate PR to be merged first